hazelcast/src/main/java/com/hazelcast/aws/AwsRequestUtils.java

```diff
@@ -47,8 +47,7 @@ static String currentTimestamp(Clock clock) {
 
     static RestClient createRestClient(String url, AwsConfig awsConfig) {
         return RestClient.create(url)
-            .withConnectTimeoutSeconds(awsConfig.getConnectionTimeoutSeconds())
-            .withReadTimeoutSeconds(awsConfig.getReadTimeoutSeconds())
+            .withPreferredTimeoutSeconds(awsConfig.getConnectionTimeoutSeconds(), awsConfig.getReadTimeoutSeconds())
```

Comment:
Due to the AWS config, It probably makes sense to have both - the `timeout` and the `connectionTimeout` (using `HttpClient.Builder.connectTimeout(Duration)`). If we want to support just one, we should improve the AWS config too.

Reply:
Good point, thanks @kwart - I had originally dropped separate values since they seemed to be the same in all areas of the codebase, but this doesn't account for customizable values such as an AWS config. I've restored functionality for both timeouts in this commit 👍 https://github.com/hazelcast/hazelcast/pull/25654/commits/6ea8e09c4534dfb8c387107d2dd6a87d655c6ddb